docs/hosting/external-storage/s3.md

```diff
@@ -0,0 +1,60 @@
+---
+contentType: howto
+---
+
+# Storing n8n's binary data in S3
+
+n8n can use [AWS S3](https://docs.aws.amazon.com/AmazonS3/latest/userguide/Welcome.html) as an external store for binary data produced by workflow executions. Other S3-compatible services like Cloudflare R2 and Backblaze B2 may work, but are not officially supported.
+
+!!! info "Enterprise-tier feature"
+    You will need an [Enterprise license key](/enterprise-key/) for external storage. If your license key expires and you remain on S3 mode, the instance will be able to read from, but not write to, the S3 bucket.
+
+## Setup
+
+First, create and configure a bucket following the [official AWS documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-bucket.html). You can use the following policy, replacing `<bucket-name>` with the name of the bucket you created.
+
+```json
+{
+	"Version": "2012-10-17",
+	"Statement": [
+		{
+			"Sid": "VisualEditor0",
+			"Effect": "Allow",
+			"Action": ["s3:*"],
+			"Resource": ["arn:aws:s3:::<bucket-name>", "arn:aws:s3:::<bucket-name>/*"]
+		}
+	]
+}
+```
+
+Once you finish creating the bucket, you will have a host, bucket name and region, and an access key ID and secret access key. You will need to set them in n8n's environment.
+
+```sh
+export N8N_EXTERNAL_STORAGE_S3_HOST=...
```

Comment:
Should we provide an example of the values for the host at least? Just to make it easier for people to set this up and avoid trial and errors.

Reply:
Done